### DISCUSSION TITLE
CKB Dev Log 2023-12-13

### DISCUSSION THREAD

**Discussion by anipenguin** [nervosnetwork/ckb#4262] [Dev Log]
![sunrise-in-colorful-danxia](https://github.com/nervosnetwork/ckb/assets/112425356/b0fda411-f5d8-48f3-9182-2b9cbcc6e080)
_Sunrise at Zhangye Danxia Geopark_ 
📸 by @contrun

<h3>Updates</h3>
<p><em><strong>Features</strong></em></p>
<ul>
<li><strong>DApp Interoperability</strong>
<ul>
<li>Introducing <a href="https://github.com/cryptape/ckb-transaction-cobuild-poc">ckb-transaction-cobuild-poc</a>, enabling the adoption of a transaction co-build protocol and message in CKB, similar to EIP-712. This feature includes a redesigned witnesses layout, simplifying the signing process and boosting DApp interoperability.</li>
</ul>
</li>
<li><strong>Bootloader for <a href="https://github.com/nervosnetwork/ckb-vm">CKB-VM</a> Programs</strong>
<ul>
<li>We've developed <a href="https://github.com/xxuejie/ckb-stack-reorg-bootloader">ckb-stack-reorg-bootloader</a>, a program ensuring stack and program heap memory are consistently situated in separate, non-overlapping regions. This optimization allows maximum heap memory utilization while maintaining program safety.</li>
</ul>
</li>
<li><strong>Accelerated Block Syncing</strong>
<ul>
<li>Block synchronization speed has significantly improved through <a href="https://github.com/nervosnetwork/ckb/pull/4254">Hardcoding a Default <strong><code>assume_valid_target</code></strong></a>and
<a href="https://github.com/nervosnetwork/ckb/pull/3958">asynchronous block download</a></li>
<li>Benchmark results on an AWS Singapore c5.xlarge (4c8g 200G SSD) VPS demonstrate notable enhancements:</li>
</ul>
</li>
</ul>

Version | Sync time | Compared v0.111.0
-- | -- | --
v0.111.0 | 25.1h | - | 
v0.111.0 + assume_valid_target | 16.2h | Shortened 35.4%
v0.111.0 + asynchronouos block download | 21.4h | Shortened 14.7%
v0.111.0 + assume_valid_target + asynchronous block download | 8.2h | Shortened 67%

![Untitled](https://github.com/nervosnetwork/ckb/assets/112425356/2779a5f3-1e89-4a7a-abd8-22a93ec3d07f)

<p><em><strong>Improvements</strong></em></p>
<ul>
<li><strong>CKB Code Text Refinement</strong>
<ul>
<li>We've streamlined and updated textual information produced by CKB nodes during startup and operation, covering four main categories:
<ul>
<li>Command line information for starting CKB, such as sub-command about and help</li>
<li>Console messages: standard output/error streams</li>
<li>Five levels of log records</li>
<li>Custom error object text displays</li>
</ul>
</li>
<li>With assistance from <a href="https://github.com/nervosnetwork/ckb/pull/4189">the text optimizer</a>, our language in CKB logs and other messages has been polished for clarity and effectiveness.</li>
</ul>
</li>
<li><strong>CKB-Auth Examples</strong>
<ul>
<li>Check out <a href="https://github.com/nervosnetwork/ckb-auth-examples">ckb-auth-examples</a> to explore the functionality of ckb-auth</li>
</ul>
</li>
<li><strong>Exec Syscall in CKB-Auth</strong>
<ul>
<li><a href="https://github.com/nervosnetwork/ckb-auth/pull/25">ckb-auth-rs</a> now supports the exec syscall, allowing users to call ckb-auth more effectively</li>
<li><a href="https://github.com/nervosnetwork/ckb-auth/pull/25">https://github.com/nervosnetwork/ckb-auth/pull/25</a></li>
</ul>
</li>
<li><strong>Docker Image for Aarch64</strong>
<ul>
<li>Added CKB docker image for Aarch64</li>
<li><a href="https://github.com/nervosnetwork/ckb-docker-builder/pull/18">https://github.com/nervosnetwork/ckb-docker-builder/pull/18</a></li>
</ul>
</li>
</ul>
<h3>In Pipeline…</h3>
<ul>
<li><strong>Staging Environment</strong>
<ul>
<li>Discussions are ongoing to establish a staging environment. This will enable network nodes to test changes in CKB with minimal disruption.</li>
</ul>
</li>
<li><strong>CKB 0.112.0 Improvement - Replace-by-Fee (RBF)</strong>
<ul>
<li>The update in CKB 0.112.0 includes the addition of RBF to the transaction pool. We're preparing a manual to assist node operators with diagnostics and transaction pool issues related to RBF.</li>
</ul>
</li>
<li><strong>CKB-VM Debugging Enhancements</strong>
<ul>
<li>Plans are in place to improve the debugging experience for CKB-VM. This includes more specific error messages to help developers pinpoint memory access issues.</li>
</ul>
</li>
</ul>
<h3>Calling for your ideas!</h3>
<p>Join the discussion started by @contrun on improving CKB's P2P network at  <a href="https://talk.nervos.org/t/ckb-s-p2p-network-update-a-call-for-community-ideas-for-enhancements/7541">CKB’s P2P Network Update: A Call for Community Ideas for Enhancements</a></p>

**Comment by roughpandaz** [comment]
@anipenguin Are these dev-logs posted on twitter somewhere? Or on any other places?

**Reply by anipenguin** [reply]
ckbdev usually post about it on twitter and i think couple of the community discord/telegram servers have bots that share it
